ethereum/statetransition/src/main/java/tech/pegasys/artemis/statetransition/util/SlotProcessorUtil.java

```diff
@@ -28,20 +26,15 @@
   public static void updateBlockRoots(BeaconState state, Bytes32 previous_block_root)
       throws SlotProcessingException {
 
-    if (state.getSlot().compareTo(UnsignedLong.valueOf(Constants.GENESIS_SLOT)) > 0) {
+    if (state.getSlot() > Constants.GENESIS_SLOT) {
       List<Bytes32> latest_block_roots = state.getLatest_block_roots();
 
       latest_block_roots.set(
-          toIntExact(state.getSlot().intValue() - 1) % Constants.LATEST_BLOCK_ROOTS_LENGTH,
-          previous_block_root);
+          ((int) state.getSlot() - 1) % Constants.LATEST_BLOCK_ROOTS_LENGTH, previous_block_root);
```

Comment:
This is a problem.  Slot's will overflow int so we will need to find a way to start slot from 0, but convert it back to a ulong before/after serialization.

Reply:
Switching to using `toIntExact`